src/pages/patientView/clinicalInformation/ClinicalInformationContainer.tsx

```diff
@@ -1,35 +1,37 @@
 import * as React from 'react';
 import ClinicalInformationPatientTable from './ClinicalInformationPatientTable';
 import Spinner from 'react-spinkit';
-import { actionCreators, mapStateToProps } from './duck';
+import Connector from './Connector';
 import { connect } from 'react-redux';
 import ClinicalInformationSamples from './ClinicalInformationSamplesTable';
 import PatientHeaderUnconnected from '../patientHeader/PatientHeader';
 import {ClinicalDataBySampleId} from "./getClinicalInformationData";
+import {ClinicalInformationData} from "./Connector";
 
-type TODO = any;
+let _ClinicalInformationData:ClinicalInformationData = null as any;
 
-interface IClinicalInformationContainerProps {
-    status: TODO;
-    patient: TODO;
-    samples: Array<ClinicalDataBySampleId>;
-    loadClinicalInformationTableData: void;
-    nodes:any;
-    setTab: void;
-}
+export interface IClinicalInformationContainerProps {
+    status?: typeof _ClinicalInformationData.status;
+    patient?: typeof _ClinicalInformationData.patient;
+    samples?: Array<ClinicalDataBySampleId>;
+    loadClinicalInformationTableData?: ()=>void;
+    nodes?:any;
+    setTab?: void;
+};
 
-export class ClinicalInformationContainerUnconnected extends React.Component<IClinicalInformationContainerProps, {}> {
+@Connector.decorator
+export default class ClinicalInformationContainer extends React.Component<IClinicalInformationContainerProps, {}> {
 
     componentDidMount() {
-        this.props.loadClinicalInformationTableData();
+		this.props.loadClinicalInformationTableData && this.props.loadClinicalInformationTableData();
```

Comment:
don't quite understand.  lets discuss elsehwere.  hopefully we wouldn't have to do this everywheer we call a prop method?